src/metabase/models/setting.clj

```diff
@@ -905,7 +910,8 @@
 
   ###### `:visibility`
 
-  `:public`, `:authenticated`, `:admin` (default), or `:internal`. Controls where this setting is visible
+  `:public`, `:authenticated`, `:settings-manager`, `:admin` (default), or `:internal`. Controls where this setting is
+  visible. `:settings-manager` refers to non-admins with the 'settings' permission.
```

Comment:
Just to make sure I understand this correctly. This is the new way this works after this PR, right?

| Visibility        | Who can see it?              | Who can update it?                        |
|-------------------|------------------------------|-------------------------------------------|
| :public           | The entire world             | Only admins                               |
| :authenticated    | Logged-in Users              | Only Admins                               |
| :settings-manager | Admins and Settings Managers | Admins and Settings Managers              |
| :admin            | Admins                       | Admins                                    |
| :internal         | Nobody                       | No one; usually for env-var-only Settings |

I think we're conflating who can see a Setting -- `:visibility`, which

> Controls where this setting is visible

with who can **set** a Setting. Would there ever be a situation where we'd want to let a `:settings-manager` set a Setting that should be visible to `:authenticated`? Maybe in the future `site-locale`, which has `:visibility :public`, should be settable by Settings managers. Should `:visibility` and _setability_ be two separate options?

Reply:
It's slightly different—settings managers can update settings that have `:public`, `:authenticated` or `:settings-manager` visibility. Admins can additionally update settings with `:admin` visibility.

I agree that we're conflating read access and write access in one property, but there's not really a need to have separate controls over them right now. Maybe I can just clarify the doc string if it wasn't clear (a table like the one you created might be a good idea).

Reply:
Ok, that makes sense.